### DISCUSSION TITLE
About querying and reporting numbers

### DISCUSSION THREAD

**Discussion by aj3423** [haumacher/phoneblock#114] [Q&A] [answered]
Hi;

From your previous reply:

> I have many requests to make internationalize PhoneBlock. Currently, the blocklist is tailored towards devices connected to the German telefony network, because German numbers are added with only the local prefix 0.... However the API fully understands international numbers (either with the plus prefix +49..., or the double-zero prefix 0049... Therefore, I would suggest to always use the plus format +49... for reporting numbers.

> Therefore, it is fine to lookup numbers of any country and also report spam numbers for arbitrary countries. To make PhoneBlock aware of the "target market" of a number, it would be useful to extend the voting API with information from which country the SPAM report originated. Let's make a separate issue for this.

I want to confirm that, when querying and reporting numbers, does it suport foreign numbers already? For example: +1... or +33...
You mentioned about extending the API, I guess it's not supported yet, if so, shall we start with domestic numbers and see if it works, and extend the API later?

**Comment by haumacher** [comment, marked as answer]
Yes, it does support international number format for querying and reporting. In the response however, the numbers are always displayed in a format suitable for the local German telefony network:

https://phoneblock.net/pb-test/api/num/%2B33123456789?format=json
```
[
  "PhoneInfo",
  {
    "phone": "0033123456789",
    "votes": 38,
    "rating": "B_MISSED",
    "whiteListed": false,
    "archived": false,
    "dateAdded": 1674834780000,
    "lastUpdate": 1675119600000
  }
]
```

https://phoneblock.net/pb-test/api/num/%2B49123456789?format=json
```
[
  "PhoneInfo",
  {
    "phone": "0123456789",
    "votes": 166,
    "rating": "C_PING",
    "whiteListed": false,
    "archived": false,
    "dateAdded": 1696434260367,
    "lastUpdate": 1732041329791
  }
]
```

Rating also support international format:

```
curl -X 'POST' \
  'https://phoneblock.net/pb-test/api/rate' \
  -H 'accept: text/plain; charset=utf-8' \
  -H 'Content-Type: application/json' \
  -d '{
  "phone": "+33123456789",
  "rating": "C_PING"
}'
```

Btw. there is a test system that can be used for development and test purpose: https://phoneblock.net/pb-test/api/
This system has a separate database. Querying and reporting numbers there does not interfere with the production system.

**Comment by aj3423** [comment]
Got it, thanks.

**Comment by aj3423** [comment]
I found that, it only works well with +49 numbers, other country codes will not be parsed correctly, for example:

The +49123456789 will be parsed as:
> phone: 0123456789

But for +33123456789 (France), it results in:
> phone: 0033123456789

It incudes the 33 as part of the domestic number, which means, currently the API doesn't handle international numbers other than +49.
Is this a bug?

**Reply by haumacher** [reply]
No, this correct from the perspective of the German telefony network. If you want to call `+33123456789` you normally type `0033123456789` into your phone. The `00` prefix is reserved for calls to "foreign" countries. And this is also the format international numbers are normally displayed by local devices (such as Fritz!Box routers).

**Reply by haumacher** [reply]
However, in the process of internationalizing PhoneBlock, this should either be changed to 
```
    "phone": "+33123456789",
```
or a new attribute should be added to keep the API compatible
```
    "phone": "0033123456789",
    "phoneInt": "+33123456789",
```

**Reply by aj3423** [reply]
Ah thanks, now I see why there is the leading 00. Then I'll query all numbers in this format: `00` + `country code` + `the domestic part`, such as: `0049123456789` and `0033123456789`, since this format works for all numbers.

> there is a test system that can be used for development and test purpose: https://phoneblock.net/pb-test/api/

There seems to be a bug with the test environment, it always shows 401 Unauthorized, the user credential is not working, but it works fine with the production API.

**Reply by haumacher** [reply]
Even the user database is separate on the test system, you have to register there separately.

**Reply by aj3423** [reply]
I see, thanks.
